rinja_parser/src/node.rs

```diff
@@ -204,6 +204,29 @@ impl<'a> When<'a> {
     #[allow(clippy::self_named_constructors)]
     fn when(i: &'a str, s: &State<'_>) -> ParseResult<'a, WithSpan<'a, Self>> {
         let start = i;
+        let endwhen = map(
+            consumed(ws(pair(
+                delimited(
+                    |i| s.tag_block_start(i),
+                    opt(Whitespace::parse),
+                    ws(keyword("endwhen")),
+                ),
+                cut(terminated(
+                    opt(Whitespace::parse),
+                    pair(
+                        |i| s.tag_block_end(i),
+                        many0(value((), ws(|i| Comment::parse(i, s)))),
+                    ),
+                )),
+            ))),
+            |(span, (pws, nws))| {
+                let comment = Comment {
+                    ws: Ws(pws, nws),
+                    content: "",
+                };
+                Node::Comment(WithSpan::new(comment, span))
```

Comment:
The implementation is kinda lazy, that I add a "virtual" comment node. But I guess it's good enough for the foreseeable future. :)

Reply:
Please add a comment on this code so future us won't wonder why there is a comment generated here. :laughing:

Reply:
Hehe, yeah, makes sense to write comments in a place where you would actually find them. :D